File: app/src/main/java/org/skyfaced/mvp/ui/MainView.java
package org.skyfaced.mvp.ui;

import org.skyfaced.mvp.model.ImageDto;
import org.skyfaced.mvp.ui.base.BaseView;

import java.util.List;

import moxy.viewstate.strategy.alias.AddToEndSingle;

public interface MainView extends BaseView {
    @AddToEndSingle
    void updateSimpleRecycler(List<String> items);

    @AddToEndSingle
    void updateWaifu(ImageDto image);

    @AddToEndSingle
    void updateWaifuRecycler(List<ImageDto> images);
}

File: app/src/main/java/org/skyfaced/mvp/ui/MainView.kt
package org.skyfaced.mvp.ui

import moxy.viewstate.strategy.alias.AddToEndSingle
import org.skyfaced.mvp.model.ImageDto
import org.skyfaced.mvp.ui.base.BaseView

interface MainView : BaseView {
    @AddToEndSingle
    fun updateSimpleRecycler(items: List<String>)

    @AddToEndSingle
    fun updateWaifu(image: ImageDto)

    @AddToEndSingle
    fun updateWaifuRecycler(images: List<ImageDto>)
}